noxfile.py

```diff
@@ -191,6 +191,7 @@ def maybe_create_lock(session: Session) -> bool:
             "latest",
             "--resolver-version",
             "pip-2020-resolver",
+            "--elide-unused-requires-dist",
```

Comment:
After the edits in this file, the rest of this change was generated via `rm lock.json && nox -e lock` (with attention paid to the comment in nox-support/doc-reqs.windows-amd64.lock.txt).